enhancements/bootimages.md

```diff
@@ -0,0 +1,163 @@
+---
+title: bootimage-updates
+authors:
+  - "@cgwalters"
+reviewers:
+  - "@coreos-team"
+approvers:
+  - "@coreos-team"
+creation-date: 2020-02-04
+last-updated: 2020-02-024
+status: provisional
+---
+
+# Updating bootimages
+
+This proposes a path towards integrating the bootimage into the release image and become managed by the cluster.  This aids worker scaleup speed, helps in mirroring OpenShift into disconnected environments, and allows the CoreOS team to avoid supporting in-place updates from OpenShift 4.1 into the forseeable future.
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+OpenShift will support fully clusters managing their own bootimages (in IaaS clouds), and come with tooling and documentation to aid operators on e.g. bare metal PXE to keep bootimages up date.
+
+By default in an environment managed by the [machine-api-operator](https://github.com/openshift/machine-api-operator), when a cluster is upgraded (e.g `oc adm upgrade`), the machinesets will be updated to use newer bootimages with the latest `machine-os-content`.
+
+## Motivation
+
+Most discussion of this problem originated in [this issue](https://github.com/openshift/os/issues/381).  For more background, see [OSUpgrades](https://github.com/openshift/machine-config-operator/blob/master/docs/OSUpgrades.md) which documents some of the process.
+
+Since the creation of OpenShift 4.1 and continuing until the time of this writing, there is no automated mechanism to update "bootimages" past a cluster installation.  We have a mechanism to do [in place updates](https://github.com/openshift/machine-config-operator/blob/09fe53e2e47bc6f8129376dfe389e98fc151ff48/docs/OSUpgrades.md) which has worked quite well, but there is a need to do more.
+
+### Goal 1: Scaling up workers directly into upgraded OS
+
+In an IaaS cloud with [cluster autoscaling](https://docs.openshift.com/container-platform/4.3/machine_management/applying-autoscaling.html) enabled, every worker that comes online will need to pull the latest `machine-os-content` and reboot before user workloads can land on the node.  This adds 3-4 minutes (at least) of latency to scaleup, and that time can matter significantly in "burst" scale scenarios, serverless usage, etc.
+
+### Goal 2: Avoid accumulation of technical debt across OS updates
+
+The CoreOS team must today support clusters upgraded in place (e.g. `oc adm upgrade`) from OpenShift 4.1 until the forseeable future.  We would like the ability to make potentially breaking changes, relying on the ability for the cluster to re-provision both the control plane and workers in-place from updated bootimages.
```

Comment:
It might help to explain what types of changes can't be delivered via current upgrade paths without updating the bootimages (e.g. can't fix any security issues we find in the boot process, can't update the binaries that do the initial upgrade to fix issues we've discovered or to provide new information used in the reboot and init sequences)

Reply:
I think it's not so much "can't" as "some things are painful and awkward", but I added a few examples, thanks!